packages/next/src/build/flying-shuttle/stitch-builds.ts

```diff
@@ -0,0 +1,504 @@
+        } finally {
+          copySema.release()
+        }
+      })
+    )
+  }
+  // always attempt copying not-found chunk
+  await copyPageChunk('/_not-found/page', 'app').catch(() => {})
+
+  // merge dynamic/static routes in routes-manifest
+  const [restoreRoutesManifest, currentRoutesManifest] = await Promise.all(
+    [
+      path.join(shuttleDir, 'manifests', ROUTES_MANIFEST),
+      path.join(distDir, ROUTES_MANIFEST),
+    ].map(async (f) => JSON.parse(await fs.promises.readFile(f, 'utf8')))
+  )
+  const dynamicRouteMap: Record<string, any> = {}
+  const combinedDynamicRoutes: Record<string, any>[] = [
+    ...currentRoutesManifest.dynamicRoutes,
+    ...restoreRoutesManifest.dynamicRoutes,
+  ]
+  for (const route of combinedDynamicRoutes) {
+    dynamicRouteMap[route.page] = route
+  }
+
+  const mergedRoutesManifest = {
+    ...currentRoutesManifest,
+    dynamicRoutes: getSortedRoutes(
+      combinedDynamicRoutes.map((item) => item.page)
+    ).map((page) => dynamicRouteMap[page]),
+    staticRoutes: [
+      ...currentRoutesManifest.staticRoutes,
+      ...restoreRoutesManifest.staticRoutes,
+    ],
+  }
+  await fs.promises.writeFile(
+    path.join(distDir, ROUTES_MANIFEST),
+    JSON.stringify(mergedRoutesManifest, null, 2)
+  )
+
+  // for build-manifest we use latest runtime files
+  // and only merge previous page chunk entries
+  // middleware-build-manifest.js (needs to be regenerated)
+  const [restoreBuildManifest, currentBuildManifest] = await Promise.all(
+    [
+      path.join(shuttleDir, 'manifests', BUILD_MANIFEST),
+      path.join(distDir, BUILD_MANIFEST),
+    ].map(async (file) => JSON.parse(await fs.promises.readFile(file, 'utf8')))
+  )
+  const mergedBuildManifest = {
+    // we want to re-use original runtime
+    // chunks so we favor restored version
+    // over new
+    ...currentBuildManifest,
+    pages: {
+      ...restoreBuildManifest.pages,
+      ...currentBuildManifest.pages,
+    },
+  }
+
+  // _app and _error is unique per runtime
+  // so nest under each specific entry in build-manifest
+  const internalEntries = ['/_error', '/_app']
+
+  for (const entry of Object.keys(restoreBuildManifest.pages)) {
+    if (currentBuildManifest.pages[entry]) {
+      continue
+    }
+    for (const internalEntry of internalEntries) {
+      for (const chunk of restoreBuildManifest.pages[internalEntry]) {
+        if (!restoreBuildManifest.pages[entry].includes(chunk)) {
+          mergedBuildManifest.pages[entry].unshift(chunk)
+        }
+      }
+    }
+  }
+
+  for (const entry of Object.keys(currentBuildManifest.pages)) {
+    for (const internalEntry of internalEntries) {
+      for (const chunk of currentBuildManifest.pages[internalEntry]) {
+        if (!currentBuildManifest.pages[entry].includes(chunk)) {
+          mergedBuildManifest.pages[entry].unshift(chunk)
+        }
+      }
+    }
+  }
+
+  for (const key of internalEntries) {
+    mergedBuildManifest.pages[key] = []
```

Comment:
what code requires this to be initialized to an empty array?

Reply:
Since the internal entries are unique per build they are nested under each page instead of global for all pages.